tasks/service-runner/app/service/service-runner.ts

```diff
@@ -72,63 +72,71 @@ function _getErrorMessage(logStr: string): string {
   * @param operation - The requested operation
   * @param callback - Function to call with result
   */
-export function runQueryCmrFromPull(workItem: WorkItem): Promise<ServiceResponse> {
-  const { operation, scrollID } = workItem;
-  const catalogDir = `${ARTIFACT_DIRECTORY}/${operation.requestId}/${workItem.id}/outputs`;
-  const args = [
-    'tasks/query-cmr/app/cli',
-    '--harmony-input',
-    `${JSON.stringify(operation)}`,
-    '--scroll-id',
-    scrollID,
-    '--output-dir',
-    catalogDir,
-  ];
-
-  return new Promise<ServiceResponse>((resolve) => {
-    logger.debug('CALLING WORKER');
-    // create a writable stream to capture stdout from the exec call
-    // using stdout instead of stderr because the service library seems to log ERROR to stdout
-    const stdOut = new LogStream();
-
-    // timeout if things take too long
-    const timeout = setTimeout(() => {
-      resolve({ error: `Worker timed out after ${workerTimeout / 1000.0} seconds` });
-    }, workerTimeout);
-
-    exec.exec(
-      'argo',
-      env.myPodName,
-      'worker',
-      [
-        'node',
-        ...args,
-      ],
-      stdOut,
-      process.stderr as stream.Writable,
-      process.stdin as stream.Readable,
-      true,
-      (status: k8s.V1Status) => {
-        logger.debug(`SIDECAR STATUS: ${JSON.stringify(status, null, 2)}`);
-        if (status.status === 'Success') {
-          clearTimeout(timeout);
-          logger.debug('Getting STAC catalogs');
-          const catalogs = _getStacCatalogs(`${catalogDir}`);
-          resolve({ batchCatalogs: catalogs });
-        } else {
-          clearTimeout(timeout);
-          const logErr = _getErrorMessage(stdOut.logStr);
-          const errMsg = `${env.harmonyService}: ${logErr}`;
-          stdOut.destroy();
-          resolve({ error: errMsg });
-        }
-      },
-    ).catch((e) => {
-      clearTimeout(timeout);
-      logger.error(e.message);
-      resolve({ error: e.message });
+export async function runQueryCmrFromPull(workItem: WorkItem): Promise<ServiceResponse> {
+  try {
+    const { operation, scrollID } = workItem;
+    const catalogDir = `${ARTIFACT_DIRECTORY}/${operation.requestId}/${workItem.id}/outputs`;
+    const args = [
+      'tasks/query-cmr/app/cli',
+      '--harmony-input',
+      `${JSON.stringify(operation)}`,
+      '--scroll-id',
+      scrollID,
+      '--output-dir',
+      catalogDir,
+    ];
+
+    return new Promise<ServiceResponse>((resolve) => {
+      logger.debug('CALLING WORKER');
+      // create a writable stream to capture stdout from the exec call
+      // using stdout instead of stderr because the service library seems to log ERROR to stdout
+      const stdOut = new LogStream();
+
+      // timeout if things take too long
+      const timeout = setTimeout(() => {
+        resolve({ error: `Worker timed out after ${workerTimeout / 1000.0} seconds` });
+      }, workerTimeout);
+
+      exec.exec(
```

Comment:
It's been too long since I had to worry about promises like this: If this method throws synchronously, does the exception get automatically propagated to the `.catch`?

Reply:
As I understand it `.catch` is just handling rejected promises, so if the `exec` call fails synchronously it will bubble up to the outer `try/catch` and be handled there.